src/shared/openssl-util.h

```diff
@@ -161,8 +172,35 @@ static inline int openssl_load_key_from_token(
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(X509*, X509_free, NULL);
 DEFINE_TRIVIAL_CLEANUP_FUNC_FULL(EVP_PKEY*, EVP_PKEY_free, NULL);
 
+struct AskPasswordUserInterface {
```

Comment:
maybe prefix this with "OpenSSL"? i.e. make this OpenSSLAskPasswordUI?

seems so misleadingly super generic otherwise.